test/Core/LeanCode.Components.Startup.Tests/LeanStartupTests.cs

```diff
@@ -0,0 +1,91 @@
+using LeanCode.Components;
+using LeanCode.Components.Startup;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Xunit;
+
+namespace LeanCode.Startup.Tests;
+
+public class LeanStartupTests
+{
+    [Fact]
+    public async Task Sink_closes_when_we_start_the_app_with_closeAndFlushLogger_parameter_set_as_true()
+    {
+        var host = LeanProgram.BuildMinimalHost<TestStartupWithTrueParameter>().ConfigureDefaultLogging("test").Build();
+
+        await host.StartAsync();
+
+        var lifetime = host.Services.GetService<IHostApplicationLifetime>();
+        lifetime.StopApplication();
+
+        await host.WaitForShutdownAsync();
+
+        var logger = host.Services.GetService<ILogger<TestStartupWithTrueParameter>>();
+
+        var isSinkClosed =
+            !logger.IsEnabled(LogLevel.Debug)
+            && !logger.IsEnabled(LogLevel.Information)
+            && !logger.IsEnabled(LogLevel.Warning)
+            && !logger.IsEnabled(LogLevel.Error)
+            && !logger.IsEnabled(LogLevel.Critical);
+
+        Assert.True(isSinkClosed);
```

Comment:
This can be done simpler - we know that we use Serilog and we know how it works, we can [use that knowledge](https://github.com/serilog/serilog/blob/f72a7a328d38f559632ae64aba08dfcf0b1dae44/src/Serilog/Log.cs#L51-L56) to make this assert:

```suggestion
        Assert.Same(Log.Logger, Serilog.Core.Logger.None);
```